src/NServiceBus.Core/Routing/MessageDrivenSubscriptions/TypePublisherSource.cs

```diff
@@ -34,10 +34,7 @@ public IEnumerable<PublisherTableEntry> GenerateWithoutBestPracticeEnforcement(C
             {
                 throw new Exception($"Cannot configure publisher for type '{messageType.FullName}' because it is not considered a message. Message types have to either implement NServiceBus.IMessage interface or match a defined message convention.");
             }
-            if (conventions.IsCommandType(messageType))
-            {
-                throw new Exception($"Cannot configure publisher for type '{messageType.FullName}' because it is a command.");
-            }
+
```

Comment:
Agreed, given our new line of thinking that best practices should we enforces where they make sense I'm no longer sure we should remove this at all. I have reverted the change and made the relevant tests only run for message driven pubsub since that avoids the startup checks